File: Algorithms/Py_Algos/BubbleSort.py
from random import randint

def create_array(h, size):
    return [randint(0, h) for _ in range(size)]

def bubbleSort(arr:list[int]) -> list[int]:
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            try:
                if arr[j] > arr[j+1]:
                    tmp = arr[j]
                    arr[j] = arr[j+1]
                    arr[j+1] = tmp
            except IndexError:
                pass

    return arr


if __name__ == "__main__":
    arr = create_array(100, 10)
    arr = bubbleSort(arr)
    print(arr)

File: Algorithms/BubbleSort_Go/BubbleSort.go
package main
import (
  "fmt"
  "math/rand"
  )

func create_array(h int, size int) []int{
  array := make([]int, size)

  for i:= 0; i < size; i++ {
    array[i] = rand.Intn(h +1)
  }

  return array
}

func bubbleSort(arr []int) []int{
  n := len(arr)
  for i:=0; i< n; i++{
    for j:=0; j< (n - 1 - i); j++{
      if arr[j] > arr[j+1] {
        tmp := arr[j]
        arr[j] = arr[j+1]
        arr[j+1] = tmp
      }
    }
  }
  return arr
}

func main() {
  arr := create_array(10, 10)
  fmt.Println(arr)
  arr2 := bubbleSort(arr)
  fmt.Println(arr2)
}
